pkg/gateway/gateway.go

```diff
@@ -0,0 +1,165 @@
+package gateway
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/zos/pkg"
+	"github.com/threefoldtech/zos/pkg/network/namespace"
+	"github.com/threefoldtech/zos/pkg/network/types"
+	"github.com/threefoldtech/zos/pkg/zinit"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	traefikService = "traefik"
+)
+
+type gatewayModule struct {
+	proxyConfigPath string
+	traefikStarted  bool
+}
+
+type ProxyConfig struct {
+	Http HTTPConfig
+}
+
+type HTTPConfig struct {
+	Routers  map[string]Router
+	Services map[string]Service
+}
+
+type Router struct {
+	Rule    string
+	Service string
+}
+
+type Service struct {
+	LoadBalancer LoadBalancer
+}
+
+type LoadBalancer struct {
+	Servers []Server
+}
+
+type Server struct {
+	Url string
+}
+
+func New(root string) pkg.Gateway {
+	configPath := filepath.Join(root, "proxy")
+	// where should service-restart/node-reboot recovery be handled?
+	if _, err := os.Stat(configPath); os.IsNotExist(err) {
+		err := os.MkdirAll(configPath, 0644)
+		if err != nil {
+			// ok to panic?
+			panic(errors.Wrap(err, "couldn't make gateway config dir"))
+		}
+	}
+
+	g := &gatewayModule{
+		proxyConfigPath: configPath,
+		traefikStarted:  false,
+	}
+	log.Debug().Bool("exist", namespace.Exists(types.PublicNamespace)).Msg("checking public namespace")
+	if !isTraefikStarted() && namespace.Exists(types.PublicNamespace) {
+		err := g.startTraefik()
+		if err != nil {
+			log.Error().Err(err).Msg("couldn't start traefik")
+		}
+	}
+	return g
+}
+func isTraefikStarted() bool {
+	z, err := zinit.New("")
+	if err != nil {
+		panic(errors.Wrap(err, "couldn't get zinit client"))
+	}
+	defer z.Close()
+	started := true
+	traefikStatus, err := z.Status(traefikService)
+	if err != nil {
+		started = false
+	}
+	log.Debug().Str("state", traefikStatus.State.String()).Msg("checking traefik state")
+	started = traefikStatus.State.String() == zinit.ServiceStateRunning
```

Comment:
```suggestion
	started = traefikStatus.State == zinit.ServiceStateRunning
```

Reply:
the compiler complains with this form. There's `.Is` method in state.